scipy/sparse/tests/test_construct.py

```diff
@@ -473,3 +473,34 @@ def test_random_accept_str_dtype(self):
         # for the dtype
         a = construct.random(10, 10, dtype='d')
 
+    def test_random_density_estimation_probabilistic(self):
+        with assert_raises(ValueError, match="expected density < 1"):
+            construct.random(20, 100, density=1,
+                             density_estimation="probabilistic")
+
+        with assert_raises(ValueError,
+                           match="density_estimation.*expected one of"):
+            construct.random(20, 100, density_estimation="unknown")
+
+        n, m = 20, 1000
+        density = 0.01
+        x1 = construct.random(n, m, density, random_state=0)
+        x2 = construct.random(n, m, density, random_state=0,
+                              density_estimation="probabilistic")
+        assert x1.shape == x2.shape == (n, m)
+        assert x1.format == x2.format
+        assert x1.dtype == x2.dtype
+        assert_allclose(x1.nnz, x2.nnz, rtol=0.1)
```

Comment:
The actually needed tolerance is something like 3e-2 on this example on my laptop. However since with a fixed random state we can get different results on different architectures (right?), I have increased so it doesn't fail later on.